src/agent/reservation.go

```diff
@@ -0,0 +1,157 @@
+package agent
+
+import (
+	"github.com/kittycash/teller/src/box"
+	"time"
+	"github.com/go-errors/errors"
+)
+
+// @TODO handle reservation expiry
+// @TODO prevent reservation by two people at the same time
+
+var (
+	ErrMaxReservationsExceeded = errors.New("User has exceeded the max number of reservations")
+	ErrBoxAlreadyReserved = errors.New("Box already reserved")
+	ErrInvalidCoinType = errors.New("Invalid coin type")
+	ErrReservationNotFound = errors.New("Reservation not found")
+	ErrInvalidReservationType = errors.New("Invalid reservation type")
+)
+
+const (
+	// Available reservation
+	Available  = "available"
+	// Reserved reservation
+	Reserved  = "reserved"
+)
+
+// Reservation is a reservation instance for a kitty box
+type Reservation struct {
+	// Address where the buyer should send the payment
+	DepositAddress string
+	// The kitty box of this reservation
+	Box *box.Box
```

Comment:
Just wondering on the thinking for having a `box.Box` type? Aren't all kitties represented by KittyID anyway so we can use that? 

https://github.com/kittycash/wallet/blob/master/src/iko/kitty.go#L10

Reply:
every kitty will be inside a box so just following that analogy
`Box` just contains the `KittyID` only. 
Also easier to make changes since we are still deciding alot of things

Reply:
Okay sweet.